timemachine/fe/dummy.py

```diff
@@ -148,43 +154,62 @@ def get_angle_anchors(bond_anchor):
     # to an independent choice of an angle anchor atom for each dummy group
     anchored_dummy_group_assignments = (
         dict(anchored_dummy_group)
-        for dummy_group_assignment in dummy_group_assignments
         for anchored_dummy_group in product(
             *[
                 [(bond_anchor, (angle_anchor, dummy_group)) for angle_anchor in get_angle_anchors(bond_anchor)]
-                for bond_anchor, dummy_group in dummy_group_assignment.items()
+                for bond_anchor, dummy_group in dummy_groups.items()
             ]
         )
     )
 
     return anchored_dummy_group_assignments
 
 
+def compute_disabled_bonds_in_dga(bond_graph, core_atoms, dga):
+    disabled_bonds = set()
+    # bond_mask = np.ones(bond_graph.number_of_edges(), dtype=np.int32)
+    atom_membership = np.zeros(bond_graph.number_of_nodes(), dtype=np.int32) - 1
+    for jdx, (anchor, dg) in enumerate(dga.items()):
+        atom_membership[anchor] = jdx
+        for da in dg:
+            atom_membership[da] = jdx
+
+    for src, dst in bond_graph.edges():
+        # src and dst are both cores -> always on
+        if src in core_atoms and dst in core_atoms:
+            pass  # always allowed
+        elif atom_membership[src] != atom_membership[dst]:
+            disabled_bonds.add(canonicalize_bond((src, dst)))
+
+    return disabled_bonds
+
+
+def compute_disabled_bonds_in_core(bond_graph_a, bond_graph_b, core_atoms_a, atom_map_a_to_b):
+    disabled_bonds = set()
+    for src_a, dst_a in bond_graph_a.edges():
+        if src_a in core_atoms_a and dst_a in core_atoms_a:
+            src_b, dst_b = atom_map_a_to_b[src_a], atom_map_a_to_b[dst_a]
+            if not bond_graph_b.has_edge(src_b, dst_b):
+                # print("missing bonds", src_a, dst_a)
+                disabled_bonds.add(canonicalize_bond((src_a, dst_a)))
+
+    return disabled_bonds
```

Comment:
[uncertainty] I'm not confident in reviewing this stretch -- e.g. I can't be very sure from reading whether there should also be a similar loop in the opposite direction, iterating over `(src_b, dst_b) in bond_graph_b.edges()` (and if `core_atoms_b, atom_map_b_to_a` should be added to the signature).

However, a 1-line docstring or line comment may prevent worry from arising about this stretch in future reads. (Also, looks closely related to `mcgregor.py::_verify_core_impl`, unsure if these can be consolidated in a future pass.)

Reply:
This should be resolved by the new approach introduced in https://github.com/proteneer/timemachine/pull/1356/commits/546037d42cb12ff9af3303002f77d09c122c2ef9, where we use the same code path for both the MCS check and `SingleTopology` assertions (which does consider the full hybrid mol, i.e. dummy atoms for the A -> B and B -> A transformations simultaneously)